app/code/Magento/Catalog/Block/Product/View/Gallery.php

```diff
@@ -170,10 +171,110 @@ public function getGalleryImagesJson()
         return json_encode($imagesItems);
     }
 
+    /**
+     * Retrieve gallery options in JSON format
+     *
+     * @return string
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     * @SuppressWarnings(PHPMD.NPathComplexity)
+     */
+    public function getGalleryOptionsJson()
+    {
+        $optionItems = null;
+
+        //Special case for gallery/nav which can be the string "thumbs/false/dots"
+        is_bool($this->getVar("gallery/nav")) ?
+            $optionItems['nav'] = $this->getVar("gallery/nav") ? 'true' : 'false' :
+            $optionItems['nav'] = $this->escapeHtml($this->getVar("gallery/nav"));
+
+        $optionItems['loop'] = $this->getVar("gallery/loop");
+        $optionItems['keyboard'] = $this->getVar("gallery/keyboard");
+        $optionItems['arrows'] = $this->getVar("gallery/arrows");
+        $optionItems['allowfullscreen'] = $this->getVar("gallery/allowfullscreen");
+        $optionItems['showCaption'] = $this->getVar("gallery/caption");
+        $optionItems['width'] = (int)$this->escapeHtml($this->getImageAttribute('product_page_image_medium', 'width'));
+        $optionItems['thumbwidth'] = (int)$this->escapeHtml(
+            $this->getImageAttribute('product_page_image_small', 'width')
+        );
+
+        if ($this->getImageAttribute('product_page_image_small', 'height') ||
+            $this->getImageAttribute('product_page_image_small', 'width')) {
+            $optionItems['thumbheight'] = (int)$this->escapeHtml(
+                $this->getImageAttribute('product_page_image_small', 'height') ?:
+                $this->getImageAttribute('product_page_image_small', 'width')
+            );
+        }
+
+        if ($this->getImageAttribute('product_page_image_medium', 'height') ||
+            $this->getImageAttribute('product_page_image_medium', 'width')) {
+            $optionItems['height'] = (int)$this->escapeHtml(
+                $this->getImageAttribute('product_page_image_medium', 'height') ?:
+                $this->getImageAttribute('product_page_image_medium', 'width')
+            );
+        }
+
+        if ($this->getVar("gallery/transition/duration")) {
+            $optionItems['transitionduration'] = (int)$this->escapeHtml($this->getVar("gallery/transition/duration"));
+        }
+
+        $optionItems['transition'] = $this->escapeHtml($this->getVar("gallery/transition/effect"));
+        $optionItems['navarrows'] = $this->getVar("gallery/navarrows");
+        $optionItems['navtype'] = $this->escapeHtml($this->getVar("gallery/navtype"));
+        $optionItems['navdir'] = $this->escapeHtml($this->getVar("gallery/navdir"));
+
+        if ($this->getVar("gallery/thumbmargin")) {
+            $optionItems['thumbmargin'] = (int)$this->escapeHtml($this->getVar("gallery/thumbmargin"));
+        }
+
+        return json_encode($optionItems);
```

Comment:
Please, use `\Magento\Framework\Serialize\Serializer\Json` instead of direct call of the `json_encode ` function

Reply:
Hi sidolov, is the use of Magento\Framework\Json\EncoderInterface acceptable instead? That is already used elsewhere in the Gallery class for json encoding, so makes sense to be consistent.

Reply:
I just wanted to check what the best way forward is.

Gallery.php is a marked API and instantiates Magento\Framework\Json\EncoderInterface in its constructor.

I realise that the use of Magento\Framework\Json\EncoderInterface is deprecated, hence you suggesting I use \Magento\Framework\Serialize\Serializer\Json instead, but the class already instantiates a Magento\Framework\Json\EncoderInterface object as $jsonEncoder which is used in the existing functions getMagnifier and getBreakpoints. 

It makes sense to use the existing Magento\Framework\Json\EncoderInterface object even though it is deprecated.

What do you think?